integrations/access/accesslist/app_test.go

```diff
@@ -0,0 +1,203 @@
+	})
+
+	accessList, err := accesslist.NewAccessList(header.Metadata{
+		Name: "test-access-list",
+	}, accesslist.Spec{
+		Title:  "test access list",
+		Owners: []accesslist.Owner{{Name: "owner1"}, {Name: "not-found"}},
+		Grants: accesslist.Grants{
+			Roles: []string{"role"},
+		},
+		Audit: accesslist.Audit{
+			NextAuditDate: clock.Now().Add(28 * 24 * time.Hour), // Four weeks out from today
+			Notifications: accesslist.Notifications{
+				Start: time.Hour * 24 * 14, // Start alerting at two weeks before audit date
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	// No notifications for today
+	advanceAndLookForRecipients(t, bot, as, clock, 0, accessList)
+
+	// Advance by one week, expect no notifications.
+	advanceAndLookForRecipients(t, bot, as, clock, 24*7*time.Hour, accessList)
+
+	// Advance by one week, expect a notification. "not-found" will be missing as a recipient.
+	advanceAndLookForRecipients(t, bot, as, clock, 24*7*time.Hour, accessList, "owner1")
+
+	// Add a new owner.
+	accessList.Spec.Owners = append(accessList.Spec.Owners, accesslist.Owner{Name: "owner2"})
+
+	// Advance by one day, expect a notification only to the new owner.
+	advanceAndLookForRecipients(t, bot, as, clock, 24*time.Hour, accessList, "owner2")
+
+	// Advance by one day, expect no notifications.
+	advanceAndLookForRecipients(t, bot, as, clock, 24*time.Hour, accessList)
+
+	// Advance by five more days, to the next week, expect two notifications
+	advanceAndLookForRecipients(t, bot, as, clock, 24*5*time.Hour, accessList, "owner1", "owner2")
+
+	// Advance by one day, expect no notifications
+	advanceAndLookForRecipients(t, bot, as, clock, 24*time.Hour, accessList)
+
+	// Advance by one day, expect no notifications
+	advanceAndLookForRecipients(t, bot, as, clock, 24*time.Hour, accessList)
+
+	// Advance by five more days, to the next week, expect two notifications
+	advanceAndLookForRecipients(t, bot, as, clock, 24*5*time.Hour, accessList, "owner1", "owner2")
+
+	// Advance by one year a week at a time, expect two notifications each time.
+	for i := 0; i < 52; i++ {
+		advanceAndLookForRecipients(t, bot, as, clock, 24*7*time.Hour, accessList, "owner1", "owner2")
+	}
+}
+
+func advanceAndLookForRecipients(t *testing.T,
+	bot *mockMessagingBot,
+	alSvc services.AccessLists,
+	clock clockwork.FakeClock,
+	advance time.Duration,
+	accessList *accesslist.AccessList,
+	recipients ...string) {
+	t.Helper()
+
+	ctx := context.Background()
+
+	_, err := alSvc.UpsertAccessList(ctx, accessList)
+	require.NoError(t, err)
+
+	bot.lastReminderRecipients = nil
+
+	var expectedRecipients []recipient.Recipient
+	if len(recipients) > 0 {
+		expectedRecipients = make([]recipient.Recipient, len(recipients))
+		for i, r := range recipients {
+			expectedRecipients[i] = recipient.Recipient{Name: r}
+		}
+	}
+	clock.Advance(advance)
+
+	require.EventuallyWithT(t, func(collect *assert.CollectT) {
+		require.ElementsMatch(collect, expectedRecipients, bot.lastReminderRecipients)
```

Comment:
```suggestion
	require.EventuallyWithT(t, func(t *assert.CollectT) {
		assert.ElementsMatch(t, expectedRecipients, bot.lastReminderRecipients)
```

This makes it impossible to mistakenly assert against `t` instead of `collect`.